src/commands/xkcd.ts

```diff
@@ -0,0 +1,144 @@
+import { ApplicationCommandOptionType, EmbedBuilder } from 'discord.js';
+import axios from 'axios';
+
+interface GetComicResponse {
+	month: string;
+	num: number;
+	link: string;
+	news: string;
+	safe_title: string;
+	transcript: string;
+	alt: string;
+	year: string;
+	title: string;
+	day: string;
+	img: string;
+}
+
+// a quick call to see the latest comic.
+// @return the data of the most recent comic, or an empty response with num set to -1 to indicate an error.
+async function _latestCheck(): Promise<GetComicResponse> {
+	try {
+		const { data, status } = await axios.get<GetComicResponse>('https://xkcd.now.sh/?comic=latest');
+		if (status !== 200) throw new Error(`${status}`);
+		return data;
+	} catch (error_) {
+		console.log(error_);
+		const error: GetComicResponse = {
+			month: '',
+			num: -1,
+			link: '',
+			news: '',
+			safe_title: '',
+			transcript: '',
+			alt: '',
+			year: '',
+			title: '',
+			day: '',
+			img: '',
+		};
+		return error;
+	}
+}
+
+export const xkcd: GlobalCommand = {
+	name: 'xkcd',
+	description: 'Fetches the most recent xkcd comic, or a selected one.',
+	requiresGuild: false,
+	options: [
+		{
+			name: 'number',
+			description: 'The index number of the comic you would like to have',
+			type: ApplicationCommandOptionType.Integer,
+		},
+	],
+
+	// entry point for command execution
+	async execute({ options, reply, sendTyping }) {
+		// sends an ephemeral message to the user
+		// @param reply: the reply method given in the execute method's CommandContext
+		// @param content: the string that should be sent to the client to let them know the error.
+		const _sendErr = async (content: string): Promise<void> => {
+			await reply({ content: content, ephemeral: true });
+		};
```

Comment:
What's your opinion on the `function` keyword? I think the main difference is the scoping of `this`, but that mostly matters for class instances, which aren't used here. I personally prefer `function`'s syntax over arrow functions when the function is more than a line or two, unless the function is being passed directly as a a callback to another function, but that's me.

```suggestion
		async function _sendErr(content: string): Promise<void> {
			await reply({ content: content, ephemeral: true });
		};
```

Reply:
Done! I personally prefer only using `function` when it's by itself (rather than inside of another function, where I prefer lambdas in most instances), but I don't have a preference that can't be changed.